File: Semestre-1/Cours-POO/Examen-Final_2022-2023/Exercice-2_Gestion-Employes/Implementation-Java/Patron.java
import java.util.ArrayList;
import java.util.List;

public class Patron extends Employe {
    private double primeRisque;
    private List<Employe> subordonnes;

    public Patron(String numeroSecuriteSociale, String nom, double salaire, Employe superieur, double primeRisque) {
        super(numeroSecuriteSociale, nom, salaire, "Patron", superieur);
        this.primeRisque = primeRisque;
        this.subordonnes = new ArrayList<>();
    }

    public void ajouterSubordonne(Employe subordonne) {
        subordonnes.add(subordonne);
    }

    @Override
    public double calculerSalaire() {
        return salaire + primeRisque;
    }

    @Override
    public String afficherSubordonnes() {
        String sub = super.afficherSubordonnes() + "\n";
        for (Employe employe : subordonnes)
            sub += "\t\t" + employe.afficheEmploye() + "\n";
        return sub;
    }
}

File: Semestre-1/Cours-POO/Examen-Final_2022-2023/Exercice-2_Gestion-Employes/Implementation-Cpp/Patron.cpp
#include "Patron.h"

Patron::Patron(string numeroSecuriteSociale, string nom, double salaire, Employe * superieur, double primeRisque) : Employe(numeroSecuriteSociale, nom, salaire, "Patron", superieur) {
    this->primeRisque = primeRisque;
}

void Patron::ajouterSubordonne(Employe * subordonne) {
    subordonnes.push_back(subordonne);
}

double Patron::calculerSalaire() const {
    return salaire + primeRisque;
}

string Patron::afficherSubordonnes() const {
    string sub = Employe::afficherSubordonnes() + "\n";
    for (const Employe * employe : subordonnes) {
        sub += "\t\t" + employe->afficheEmploye() + "\n";
    }
    return sub;
}